src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
:+1: for multiple serializers
We encrypt some cardholder data before send it to the Grain State and would be cool if I can ser a custom serializer just for that kind of data so we don't need to implement this code inside the grain.